pgcli/packages/pgspecial.py

```diff
@@ -25,6 +25,26 @@ def parse_special_command(sql):
     verbose = '+' in command
     return (command.strip(), verbose, arg.strip())
 
+def list_schemas(cur, pattern, verbose):
+    """
+    Returns (rows, headers, status)
+    """
+    sql = """SELECT n.nspname AS "Name",
+    pg_catalog.pg_get_userbyid(n.nspowner) AS "Owner"
+    FROM pg_catalog.pg_namespace n WHERE n.nspname """
+
+    if pattern:
+        sql += "~ '^(" + pattern.replace('*', '.*') + ")$'"
```

Comment:
I did base the sql_name_pattern() on the original from psql, but I guess I didn't fully cover all the cases and chose to go with a simplified impl. 

Would you like to take a stab at it or should I file an issue for it and get back to it at a later time?